libraries/aws_iam_user.rb

```diff
@@ -12,51 +12,49 @@ class AwsIamUser < AwsResourceBase
     end
   "
 
-  attr_reader :access_keys, :attached_policy_names, :attached_policy_arns,
-              :has_console_password, :has_mfa_enabled, :inline_policy_names,
-              :username, :user_arn, :user_id
-
-  alias has_mfa_enabled? has_mfa_enabled
-  alias has_console_password? has_console_password
+  attr_reader :username, :user_arn, :user_id
 
   def initialize(opts = {})
     opts = { user_name: opts } if opts.is_a?(String)
     super(opts)
     validate_parameters(required: [:user_name])
 
-    catch_aws_errors do
-      username   = { user_name: opts[:user_name] }
-      iam_client = @aws.iam_client
-
-      begin
-        # Basic User attributes
-        user = iam_client.get_user(username).user
-        @user_arn = user.arn
-        @user_id  = user.user_id
-        @username = opts[:user_name]
-
-        # Additional attributes
-        @has_console_password = has_password?(username)
-        @access_keys          = user_access_keys(username)
-        @has_mfa_enabled      = !iam_client.list_mfa_devices(username).mfa_devices.empty?
-        @inline_policy_names  = iam_client.list_user_policies(username).policy_names
-
-        policies = iam_client.list_attached_user_policies(username).attached_policies
-        @attached_policy_arns  = policies.map { |p| p[:policy_arn] }
-        @attached_policy_names = policies.map { |p| p[:policy_name] }
-      rescue Aws::IAM::Errors::NoSuchEntity
-        # If the requested User does not exist
-        @user_id = nil
-        @user_arn = nil
-
-        @access_keys = []
-        @inline_policy_names = []
-        @attached_policy_arns = []
-        @attached_policy_names = []
-      end
+    user = catch_aws_errors do
+      iam_client.get_user(user_params).user
     end
+    @user_arn = user.arn
+    @user_id  = user.user_id
+    @username = opts[:user_name]
+  end
+
+  def access_keys
+    @access_keys ||= (@aws.iam_client.list_access_keys(user_params).access_key_metadata || [])
+  end
+
+  def has_console_password
```

Comment:
I think you need to define this with a question mark at the end of the method name to maintain the semantics. Previously there was an alias, but you have deleted the alias.